### ISSUE TITLE
A rede está uma porcaria

### ISSUE THREAD

**Issue by ghost** [Ropago/EPIA#1]
>  Problema

Como pode ser visto na matriz de confusão gerada, no log do verbose e nos erros de validação a cada fold, a rede não melhora nunca. Por algum motivo o aprendizado não acontece e por isso ela fica interrompendo as épocas de treinamento, e o teste da rede (visto na matriz de confusão) se acumula tudo em uma classe só.

> Possibilidades

**1) overfitting**
Quando vc testa e treina a rede com os mesmos dados. A predição seria ótima no momento, mas ao colocar outros dados não daria em nada.
Só que o Kfold já toma conta disso, ele separa parte dos arquivos de treinamento para teste.
Além disso, depois do treinamento o teste real é feito com outros dados.

Pode ser possível que seja algo a ver com o k fold. Possibilidades para teste: Tentar o Group k-fold
Link: <http://scikit-learn.org/stable/modules/cross_validation.html#group-k-fold>

Descrição:
"For example if the data is obtained from different subjects with several samples per-subject and if the model is flexible enough to learn from highly person specific features it could fail to generalize to new subjects. class:GroupKFold makes it possible to detect this kind of overfitting situations.	"

**2) descritor não é bacana para esse tipo de imagem**
Como é um descritor voltado a textura, quanto de textura diferenciada pode ter um conjunto de letras?
Faz sentido que a textura seja a mesma para todas...

**3) geração errada do LBP**
O LBP está comprometido.
Os dados estao vindo errados?
Seria o histograma ou outra coisa?

Outro exemplo de normalização:
link <http://scikit-image.org/docs/dev/auto_examples/features_detection/plot_local_binary_pattern.html>

n_bins = int(lbp.max() + 1)

return ax.hist(lbp.ravel(), normed=True, bins=n_bins, range=(0, n_bins), facecolor='0.5')

Ou ainda:
n_bins = int(lbp.max() + 1)

ref_hist, _ = np.histogram(ref, normed=True, bins=n_bins, range=(0, n_bins))

Outro link:
http://aurlabcvsimulator.readthedocs.io/en/latest/PlotLBPsHistogram/
